lib/assert.js

```diff
@@ -931,6 +933,87 @@ assert.doesNotMatch = function doesNotMatch(string, regexp, message) {
   internalMatch(string, regexp, message, doesNotMatch);
 };
 
+const noop = () => {};
+
+class Tracker {
+
+  constructor() {
+    this.callChecks = new SafeSet();
+  }
```

Comment:
For this, I would make `callChecks` a private internal field...
```suggestion
  #callChecks = new SafeSet();
```

Then update the references throughout from `this.callChecks` to `this.#callChecks`